graphs/uniform_search_cost.py

```diff
@@ -0,0 +1,163 @@
+##################### -- Learning Source -- #####################
+
+# 1. https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm
+# 2. https://www.educative.io/answers/what-is-uniform-cost-search
+
+# In above 1 no. link, you will get about dijkstra algorithm.
+# You may have to scroll down a little bit.
+# And you will get a section titled as "Practical optimizations and infinite graphs"
+# This section explains about Uniform Cost Search algorithm.
+
+# Uniform search cost is slightly similar to dijkstra.
+# The main different between them is that UCS is having multiple goals.
+
+# Below code is fully based on graph node environment
+##################### --------------------- #####################
+
+
+def uniform_search_cost(
+    graph_data: list[list[int]],
+    distance: dict[
+        tuple[
+            int,
+            int,
+        ],
+        int,
+    ],
+    goal: list[int],
+    start: int,
+) -> list[int]:
+    """
+    Returns a list of integer values which are result of optimal distance
+    >>> graph_data = [[1, 2], [2, 3], [], []]
+    >>> distance = {(0, 1): 2, (0, 2): 1, (1, 2): 5, (1, 3): 10}
+    >>> goal = [2]
+    >>> start = 0
+    >>> uniform_search_cost(graph_data, distance, goal, start)
```

Comment:
Can you create some more test cases with negative values, 0, floats, etc?

Reply:
UCS typically assumes that edge weights are non-negative. This means that UCS may not work correctly when dealing with graphs that have negative edge weights. Because Uniform Cost Search (UCS) is a variant of Dijkstra's algorithm that is used to find the shortest path in a weighted graph. Yes, floats and 0 supports in UCS.